x-pack/plugin/mapper-version/src/main/java/org/elasticsearch/xpack/versionfield/Version.java

```diff
@@ -7,21 +7,28 @@
 
 package org.elasticsearch.xpack.versionfield;
 
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.script.BytesRefProducer;
 import org.elasticsearch.xcontent.ToXContent;
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
 
 /**
  * Script value class.
- * TODO(stu): implement {@code Comparable<Version>} based on {@code VersionEncoder#prefixDigitGroupsWithLength(String, BytesRefBuilder)}
- * See: https://github.com/elastic/elasticsearch/issues/82287
  */
-public class Version implements ToXContent {
+public class Version implements ToXContent, BytesRefProducer, Comparable<Version> {
```

Comment:
Minor nit, unrelated to this change actually, but could you change this class to implement `ToXContentFragment` and remove the `isFragment` method? We use `ToXContentFragment`  as a marker interface for things that don't output full json object, and in this case I think the class itself only outputs a value to the xcontent builder.